tools/pythonpkg/src/pyrelation/initialize.cpp

```diff
@@ -67,9 +67,15 @@ static void InitializeConsumers(py::class_<DuckDBPyRelation> &m) {
 	    .def("pl", &DuckDBPyRelation::ToPolars, "Execute and fetch all rows as a Polars DataFrame",
 	         py::arg("batch_size") = 1000000)
 	    .def("torch", &DuckDBPyRelation::FetchPyTorch, "Fetch a result as dict of PyTorch Tensors")
-	    .def("tf", &DuckDBPyRelation::FetchTF, "Fetch a result as dict of TensorFlow Tensors")
-	    .def("record_batch", &DuckDBPyRelation::ToRecordBatch,
-	         "Execute and return an Arrow Record Batch Reader that yields all rows", py::arg("batch_size") = 1000000)
+	    .def("tf", &DuckDBPyRelation::FetchTF, "Fetch a result as dict of TensorFlow Tensors");
+	const char *capsule_docs = R"(
+			Execute and return an ArrowArrayStream through the Arrow PyCapsule Interface.
+
+			https://arrow.apache.org/docs/dev/format/CDataInterface/PyCapsuleInterface.html
+		)";
+	m.def("__arrow_c_stream__", &DuckDBPyRelation::ToArrowCapsule, capsule_docs);
```

Comment:
Not entirely familiar with the syntax here, but my reading is that this method has no keyword arguments? 

It would be good to add the `requested_schema` keyword, even if you simply ignore it for now (which is fine, because the spec states that the handling of keyword is "best effort" anyway). But not having that keyword will give errors in consumers that pass that keyword (like `pyarrow.table(..)` will always do, even if it is None)

Reply:
Indeed, this does fail for pyarrow.table:
```py
import duckdb
import pyarrow as pa
import polars as pl

df = pl.DataFrame({"a": [1, 2, 3, 4], "b": ["a", "b", "c", "d"]})
con = duckdb.connect()
sql = "SELECT * from df"
query = con.query(sql)

test = pa.table(query)
```
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File /Users/kyle/tmp/duckdb/tmp3.py:[1](https://file+.vscode-resource.vscode-cdn.net/Users/kyle/tmp/duckdb/tmp3.py:1)
----> 1 test = pa.table(query)

File ~/tmp/duckdb/.venv/lib/python3.11/site-packages/pyarrow/table.pxi:6009, in pyarrow.lib.table()

TypeError: __arrow_c_stream__(): incompatible function arguments. The following argument types are supported:
    1. (self: duckdb.duckdb.DuckDBPyRelation) -> object

Invoked with: ┌───────┬─────────┐
│   a   │    b    │
│ int64 │ varchar │
├───────┼─────────┤
│     1 │ a       │
│     2 │ b       │
│     3 │ c       │
│     4 │ d       │
└───────┴─────────┘
, None
```

Reply:
I believe https://github.com/duckdb/duckdb/pull/13802/files should fix this